### PR TITLE
Update Polly to 7.1.0 and 3.0.0

### PR DESCRIPTION
**Pull request by rynowak**
@reisenberger @Eilon

### CODE REVIEW

File: eng/Dependencies.props [hunk 1]
```diff
@@ -64,8 +64,8 @@ and are generated based on the last package release.
     <LatestPackageReference Include="Microsoft.DotNet.GenAPI" Version="$(MicrosoftDotNetGenApiPackageVersion)" />
     <LatestPackageReference Include="Moq" Version="4.10.0" />
     <LatestPackageReference Include="Newtonsoft.Json" Version="11.0.2" />
-    <LatestPackageReference Include="Polly.Extensions.Http" Version="2.0.1" />
-    <LatestPackageReference Include="Polly" Version="6.0.1" />
+    <LatestPackageReference Include="Polly.Extensions.Http" Version="3.0.0" />
```
Comment (Eilon) [inline eng/Dependencies.props]: I don't see this one on NuGet.org. Is it not published yet?

Comment (rynowak) [reply eng/Dependencies.props]: It is not, we're staging this for them to update their package 😀

Comment (Eilon) [reply eng/Dependencies.props]: Cool. Well it's "approved" from my perspective. Once the target package is on NuGet.org and this change here gets checked in, Component Governance (CG) will auto-register it.

**Comment by reisenberger** [comment]
[Polly.Extensions.Http/3.0.0](https://www.nuget.org/packages/Polly.Extensions.Http/3.0.0) is now available on nuget (for use with this PR)

**Pull request merged**